Crypto/PublicKey+Apple.mm

```diff
@@ -340,12 +340,26 @@ virtual int _sign(int/*mbedtls_md_type_t*/ mbedDigestAlgorithm,
                 (id)kSecAttrIsPermanent:    @YES,
                 (id)kSecAttrLabel:          @(timestr),
             };
-            SecKeyRef publicKey, privateKey;
-            ++gC4ExpectExceptions;
-            OSStatus err = SecKeyGeneratePair((CFDictionaryRef)params, &publicKey, &privateKey);
-            --gC4ExpectExceptions;
-            checkOSStatus(err, "SecKeyGeneratePair", "Couldn't create a private key");
-
+            
+            SecKeyRef publicKey = NULL, privateKey = NULL;
+            if (@available(macOS 10.12, iOS 10.0, *)) {
+                CFErrorRef error;
+                ++gC4ExpectExceptions;
```

Comment:
gC4ExpectException is, I think, intended for tests Why here?

Reply:
Those are ported from the master branch. I do not know exactly but there are multiple places that use that in the file. My best guess is that Apple Keychain internally throws exception and without that all tests that use keychain could be annoyingly stopped when the exception was thrown internally. 

If the recommendation is to not using those, that should be done in a separate tickets in all places in the file.

Reply:
I created CBL-4532.